website/src/components/TutorialComponents/CodeBlock.tsx

```diff
@@ -121,19 +123,38 @@ function CodeEditor(props: {
         setTmpCode(code.slice()); // use copy not reference
         setCode(props.code);
         setDisabled(true);
+        setHasFocus(false);
         setAllowUndo(true);
         setTimeout(() => {
             setDisabled(props.disabled);
+            setHasFocus(true);
             setAllowUndo(false);
         }, 3000);
     }
 
     const onClickUndo = () => {
-        console.log(tmpCode)
         setCode(tmpCode);
     }
 
-    //   prismIncludeLanguages(Prism);
+
+    document.addEventListener('keydown',
```

Comment:
you are adding this listener on every render, instead use useEffect to add it and remove it.

Reply:
add event exclusively on the editor instance better?

Reply:
I tried `onkeydownCapture` on the editor instance itself but that has some race condition with the `onKeydown` handler of `use-editable` that we do not have control. Capture `keydown` from the very top level would solve the problem.

Reply:
Not so familiar with the `useEffect` approach, let me see if it produces the same behavior.

Reply:
something like this, you need to clean up when the react component gets unmounted
```
useEffect(() => {
   const listenere = () => { ... }
   document.addEventListener("keydown", listener)
   return () => document.removeEventListener("keydown", listener)
}, [])
```